ts/check.ts

```diff
@@ -1,9 +1,22 @@
 import { hasTag } from "./internal/utils.ts";
 import { NixExpression } from "./nix.ts";
 
+/**
+ * `Check`s are commands (usually shell commands) that are used to define
+ * reproducible automated tests or other checks.
+ *
+ * `Check`s are run in a sandbox. This means they don't have access to your
+ * locally installed tools, local file system, or the internet. Instead, they
+ * are always based on an underlying `Environment`, and through that will have
+ * access to the tools that the environment provides
+ *
+ * You can run `Check`s with `garn check`.
+ */
```

Comment:
I copied a lot of these from the #258 draft. Overall I think this duplication is nice since these concepts are so important that it is nice to have it show up not only on the website, but also in the deno docs and in editors with typescript LSP.